examples_notebooks/community_contrib/couchbase/GraphRAG_with_Couchbase.ipynb

```diff
@@ -0,0 +1,557 @@
+    "\n",
+    "read_indexer_entities, read_indexer_relationships, etc., are custom functions responsible for reading specific parts of the data, such as entities and relationships.\n",
+    "We use pandas to load the data from the files, and if a file is not found, we log a warning and continue."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "id": "QQTIbPrzgvyS"
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024-09-06 15:34:48,630 - __main__ - INFO - Loading data from parquet files\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024-09-06 15:34:48,805 - __main__ - WARNING - COVARIATE_TABLE file not found. Setting covariates to None.\n",
+      "2024-09-06 15:34:48,869 - __main__ - INFO - Data loading completed\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Set up logging\n",
+    "logging.basicConfig(\n",
+    "    level=logging.INFO, format=\"%(asctime)s - %(name)s - %(levelname)s - %(message)s\"\n",
+    ")\n",
+    "logger = logging.getLogger(__name__)\n",
+    "logger.info(\"Loading data from parquet files\")\n",
+    "data = {}\n",
+    "\n",
+    "# Constants\n",
+    "COMMUNITY_LEVEL = 2\n",
+    "\n",
+    "# Table names\n",
+    "TABLE_NAMES = {\n",
+    "    \"COMMUNITY_REPORT_TABLE\": \"create_final_community_reports\",\n",
+    "    \"ENTITY_TABLE\": \"create_final_nodes\",\n",
+    "    \"ENTITY_EMBEDDING_TABLE\": \"create_final_entities\",\n",
+    "    \"RELATIONSHIP_TABLE\": \"create_final_relationships\",\n",
+    "    \"COVARIATE_TABLE\": \"create_final_covariates\",\n",
+    "    \"TEXT_UNIT_TABLE\": \"create_final_text_units\",\n",
+    "}\n",
+    "\n",
+    "try:\n",
+    "    data[\"entities\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['ENTITY_TABLE']}.parquet\")\n",
+    "    entity_embeddings = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['ENTITY_EMBEDDING_TABLE']}.parquet\")\n",
+    "    data[\"entities\"] = read_indexer_entities(data[\"entities\"], entity_embeddings, COMMUNITY_LEVEL)\n",
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"ENTITY_TABLE file not found. Setting entities to None.\")\n",
+    "    data[\"entities\"] = None\n",
+    "\n",
+    "try:\n",
+    "    data[\"relationships\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['RELATIONSHIP_TABLE']}.parquet\")\n",
+    "    data[\"relationships\"] = read_indexer_relationships(data[\"relationships\"])\n",
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"RELATIONSHIP_TABLE file not found. Setting relationships to None.\")\n",
+    "    data[\"relationships\"] = None\n",
+    "\n",
+    "try:\n",
+    "    data[\"covariates\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['COVARIATE_TABLE']}.parquet\")\n",
+    "    data[\"covariates\"] = read_indexer_covariates(data[\"covariates\"])\n",
+    "except FileNotFoundError:\n",
```

Comment:
Here you can explain in what format you are keeping the data. Also mention the data while adding to couchbase like this is the format etc.

Reply:
added explaination and sample data.